recipe/no-qt-binding-msg.patch

```diff
@@ -0,0 +1,58 @@
+diff --git a/napari/__main__.py b/napari/__main__.py
+index b3b4a15b..ba3e518a 100644
+--- a/napari/__main__.py
++++ b/napari/__main__.py
+@@ -549,19 +549,40 @@ def _maybe_rerun_with_macos_fixes():
+ 
+ 
+ def main():
+-    # There a number of macOS issues we can fix with env vars
+-    # and/or relaunching a subprocess
+-    _maybe_rerun_with_macos_fixes()
+-
+-    # Prevent https://github.com/napari/napari/issues/3415
+-    # This one fix is needed _after_ a potential relaunch,
+-    # that's why it's here and not in _maybe_rerun_with_macos_fixes()
+-    if sys.platform == "darwin":
+-        import multiprocessing
+-
+-        multiprocessing.set_start_method('fork')
+-
+-    _run()
++    try:
++        # There a number of macOS issues we can fix with env vars
++        # and/or relaunching a subprocess
++        _maybe_rerun_with_macos_fixes()
++
++        # Prevent https://github.com/napari/napari/issues/3415
++        # This one fix is needed _after_ a potential relaunch,
++        # that's why it's here and not in _maybe_rerun_with_macos_fixes()
++        if sys.platform == "darwin":
++            import multiprocessing
++
++            multiprocessing.set_start_method('fork')
++
++        _run()
++    except Exception as e:
++        # qtpy is also imported in _maybe_rerun_with_macos_fixes() on macos
++        # so we need to catch the ImportError here
++        # importing the exception type also raises the exception itself!
++        # so we have to do it manually with string comparisons
++        if (
++            e.__class__.__name__ == "QtBindingsNotFoundError"
++            and e.__module__ == "qtpy"
++        ):
++            msg = (
++                "napari requires either PyQt5 or PySide2 to be installed.\n\n"
++                "If you are using pip, you can install either with:\n"
++                "  $ pip install -U napari[pyqt]\n"
++                "  $ pip install -U napari[pyside]\n\n"
```

Comment:
@jaimergp at SciPy there were lots of people on ARM Macs that were getting errors with these calls, so I suggest:

```suggestion
+                "  $ pip install -U 'napari[pyqt]'\n"
+                "  $ pip install -U 'napari[pyside]'\n"
+                "  $ pip install -U 'napari[pyqt6_experimental]'  # Apple Silicon\n"
+                "  $ pip install -U 'napari[pyside6_experimental]'  # Apple Silicon\n\n"
```

Reply:
Eek, we sure we want people going with Qt6? PySide6 won't even launch at the moment and PyQt6 has dialog issues.
IMO on arm64 macOS the safest option is to suggest conda installing pyqt5 or pyside2.

Reply:
Well I was actually thinking that we should maybe not mention pip in this patch because this will only show up if you were already using conda... thoughts?

Reply:
I'm totally fine with either removing pip or keeping pyqt6. Things work basically fine with PyQt6? And the tag has "experimental" right there in the title. Self-documenting! 😃 At any rate, @jaimergp, can you push ASAP either one of those and merge? 🙏 Thanks both!

Reply:
Some dialogs are still broken on main with pyqt6 6.5. I've got a PR in progress.